model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/authorization/resource/delegate/JpaResourceDelegateProvider.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.models.map.storage.jpa.authorization.resource.delegate;
+
+import java.util.UUID;
+
+import javax.persistence.EntityManager;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.JoinType;
+import javax.persistence.criteria.Root;
+
+import org.keycloak.models.map.authorization.entity.MapResourceEntity;
+import org.keycloak.models.map.authorization.entity.MapResourceEntityFields;
+import org.keycloak.models.map.common.EntityField;
+import org.keycloak.models.map.common.delegate.DelegateProvider;
+import org.keycloak.models.map.storage.jpa.JpaDelegateProvider;
+import org.keycloak.models.map.storage.jpa.authorization.resource.entity.JpaResourceEntity;
+
+public class JpaResourceDelegateProvider extends JpaDelegateProvider<JpaResourceEntity> implements DelegateProvider<MapResourceEntity> {
+
+    private final EntityManager em;
+
+    public JpaResourceDelegateProvider(JpaResourceEntity delegate, EntityManager em) {
+        super(delegate);
+        this.em = em;
+    }
+
+    @Override
+    public MapResourceEntity getDelegate(boolean isRead, Enum<? extends EntityField<MapResourceEntity>> field, Object... parameters) {
+        if (getDelegate().isMetadataInitialized()) return getDelegate();
+        if (isRead) {
+            if (field instanceof MapResourceEntityFields) {
+                switch ((MapResourceEntityFields) field) {
+                    case ID:
+                    case NAME:
+                    case TYPE:
+                    case OWNER:
+                    case REALM_ID:
+                    case RESOURCE_SERVER_ID:
+                        return getDelegate();
+
+                    case ATTRIBUTES:
+                        setDelegateWithAssociation("attributes");
+                        break;
+
+                    case SCOPE_IDS:
+                        setDelegateWithAssociation("scopeIds");
+                        break;
+
+                    case URIS:
+                        setDelegateWithAssociation("uris");
+                        break;
+
```

Comment:
After adding this piece, I can see some `ModelIllegalStateException`s ... need to investigate further

Reply:
I don't know exact details, but it seems it doesn't work correctly with `ElementCollection`s ... we might be changing those in future to `@OneToMany` so I removed the `ElementCollection` associations from delegates.